Validation/HGCalValidation/interface/HGCalValidator.h

```diff
@@ -96,6 +100,7 @@ class HGCalValidator : public DQMGlobalEDAnalyzer<HGCalValidatorHistograms> {
   std::unique_ptr<HGVHistoProducerAlgo> histoProducerAlgo_;
 
 private:
+  mutable TICLCandidateValidator candidateVal;
```

Comment:
Why is this `mutable`?

Reply:
because some of its members are modified in the methods of the `HGCalValidator` that are marked `const` (so far my attempts of doing this in another way have failed, but I can try again if you don't like this approach)

Reply:
The methods of `HGCalValidator` are `const` for good reason (caused by inheritance from `DQMGlobalEDAnalyzer`, and being thus constrained by thread safety and efficiency requirements of [`edm::global` modules](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface)). An unprotected `mutable` will either cause data races, or require a detailed documentation (as comments in the code) why it would be thread safe.

On a quick look it seems to me the `TICLCandidateValidator` would need similar separation of (thread-safe) modifiable histograms and `const` filling code, as is done for `HGVHistoProducerAlgoHistograms` and `HGVHistoProducerAlgo`. An alternative would be to re-engineer `HGCalValidator` as `DQMEDAnalyzer`.

Reply:
I've used the same strategy of `HGVHistoProducerAlgo` to remove the `mutable` keyword